src/type/definition.js

```diff
@@ -506,6 +506,11 @@ function defineFieldMap<TSource, TContext>(
       `${type.name}.${fieldName} field type must be Output Type but ` +
       `got: ${String(field.type)}.`
     );
+    invariant(
+      isValidResolver(field.resolve),
```

Comment:
You're right, that it would let in falsey values. We could instead use `field.resolve == null || typeof field.resolve === 'function'` to be more specific